serde/src/ser/mod.rs

```diff
@@ -616,6 +620,29 @@ pub trait Serializer: Sized {
         }
         serializer.end()
     }
+
+    /// Collect a `Display` value as a string.
+    ///
+    /// The default implementation serializes the given value as a string with
+    /// `ToString::to_string`.
+    #[cfg(any(feature = "std", feature = "collections"))]
+    fn collect_str<T>(self, value: &T) -> Result<Self::Ok, Self::Error>
+        where T: Display,
+    {
+        let mut string = String::new();
+        write!(string, "{}", value).unwrap();
```

Comment:
I would like to use `ToString`, but unfortunately, I have to [hate](https://github.com/rust-lang/rust/issues/40103) it for now.